acbuild/acbuild.go

```diff
@@ -140,7 +144,45 @@ GLOBAL OPTIONS:
 // terminator.
 func runWrapper(cf func(cmd *cobra.Command, args []string) (exit int)) func(cmd *cobra.Command, args []string) {
 	return func(cmd *cobra.Command, args []string) {
+		if aciToModify != "" {
+			contextualCommands := []string{"begin", "end", "abort"}
+			for _, cc := range contextualCommands {
+				command := strings.Split(cmd.Use, " ")[0]
+				if command == cc {
+					stderr("Can't use the --modify flag with %s.", command)
+					cmdExitCode = 1
+					return
+				}
+			}
+
+			var err error
+			contextpath, err = ioutil.TempDir("", "acbuild-")
+			if err != nil {
+				stderr("%v", err)
```

Comment:
derr, err.Error() is what I should have said. nbd.

On Wed, Oct 14, 2015 at 10:23 AM, Derek Gonyeo <notifications@github.com>
wrote:

> In acbuild/acbuild.go
> <https://github.com/appc/acbuild/pull/34#discussion_r42024052>:
>
> > @@ -140,7 +144,45 @@ GLOBAL OPTIONS:
> >  // terminator.
> >  func runWrapper(cf func(cmd *cobra.Command, args []string) (exit int)) func(cmd *cobra.Command, args []string) {
> >  	return func(cmd *cobra.Command, args []string) {
> > +		if aciToModify != "" {
> > +			contextualCommands := []string{"begin", "end", "abort"}
> > +			for _, cc := range contextualCommands {
> > +				command := strings.Split(cmd.Use, " ")[0]
> > +				if command == cc {
> > +					stderr("Can't use the --modify flag with %s.", command)
> > +					cmdExitCode = 1
> > +					return
> > +				}
> > +			}
> > +
> > +			var err error
> > +			contextpath, err = ioutil.TempDir("", "acbuild-")
> > +			if err != nil {
> > +				stderr("%v", err)
>
> Nope, the first argument to stderr must be a string.
>
> https://play.golang.org/p/2CvjwRqRwF
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/appc/acbuild/pull/34/files#r42024052>.
>